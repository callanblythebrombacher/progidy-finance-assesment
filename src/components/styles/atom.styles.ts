import {StyleSheet} from 'react-native';

export const autoCompleteStyles = StyleSheet.create({
  autoCompleteListItem: {
    backgroundColor: 'white',
    borderBottomColor: '#e7e7e7',
    borderBottomWidth: 1,
    height: 60,
  },
  autocompleteListContainer: {
    position: 'absolute',
    width: '100%',
    top: 65, // Adjust as needed based on your layout
    backgroundColor: '#fff',
    borderRadius: 5,
    elevation: 5, // Android shadow
    shadowColor: '#000', // iOS shadow
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
});

export const tableStyles = StyleSheet.create({
  tableHeader: {backgroundColor: '#dddddd', width: '100%'},
  tableRow: {height: 70, backgroundColor: 'white'},
});

export const pickerStyles = StyleSheet.create({
  container: {
    zIndex: 3,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 2,
  },
});
