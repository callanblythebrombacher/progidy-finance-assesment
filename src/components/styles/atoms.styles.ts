import {StyleSheet} from 'react-native';

export const autoCompleteStyles = StyleSheet.create({
  item: {
    backgroundColor: 'white',
    borderBottomColor: '#e7e7e7',
    borderBottomWidth: 1,
    height: 60,
  },
  container: {
    position: 'absolute',
    top: 61, // Adjust as needed based on your layout
    backgroundColor: '#fff',
    borderRadius: 5,
    elevation: 5, // Android shadow
    shadowColor: '#000', // iOS shadow
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.8,
    shadowRadius: 2,
    left: -10,
  },
});

export const tableStyles = StyleSheet.create({
  tableHeader: {backgroundColor: '#dddddd', width: '100%'},
  tableRow: {height: 70, backgroundColor: 'white'},
});

export const pickerStyles = StyleSheet.create({
  container: {
    zIndex: 3,
    marginLeft: 10,
    width: 140,
  },
});
