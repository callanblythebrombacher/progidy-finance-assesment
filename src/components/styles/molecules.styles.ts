import {StyleSheet} from 'react-native';
export const tableStyles = StyleSheet.create({
  table: {
    position: 'relative',
    zIndex: 1,
    top: 10,
  },
  scrollView: {
    backgroundColor: 'white',
  },
  deleteButtonView: {
    backgroundColor: 'red',
    width: 60,
    justifyContent: 'center',
    alignItems: 'center',
  },
  pagination: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export const searchBarStyles = StyleSheet.create({
  container: {
    backgroundColor: '#ffffff',
    width: '100%',
    zIndex: 4,
  },
  input: {
    height: 50,
    borderWidth: 1,
    borderColor: '#000000',
    padding: 10,
    marginBottom: 20,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 10,
    backgroundColor: 'white',
    borderRadius: 8,
  },
  iconButton: {
    position: 'absolute',
    right: 3,
    top: 9,
  },
});
