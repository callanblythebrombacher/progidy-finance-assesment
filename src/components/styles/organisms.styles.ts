import {StyleSheet} from 'react-native';

export const variableSearchBarStyles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    position: 'relative',
    height: 90,
    paddingTop: 20,
    paddingEnd: 10,
    paddingBottom: 20,
    paddingStart: 10,
    zIndex: 2,
  },
});
