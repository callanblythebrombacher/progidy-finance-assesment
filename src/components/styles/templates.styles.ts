import {StyleSheet} from 'react-native';

export const filterableDataTableStyles = StyleSheet.create({
  wrapper: {paddingBottom: 160, backgroundColor: '#fff'},
});
