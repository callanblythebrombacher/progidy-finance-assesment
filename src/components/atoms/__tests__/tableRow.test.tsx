import React from 'react';
import {shallow} from 'enzyme';
import {DataTable} from 'react-native-paper';
import {TableRow} from '../tableRow.atom.tsx';
import {tableStyles as styles} from '../../styles/atoms.styles.ts';

import {describe, beforeEach, it, expect, afterEach, jest} from '@jest/globals';

describe('TableRow Atom', () => {
  const mockCellData = [
    {item: 'John Doe', isNumeric: false},
    {item: '25', isNumeric: true},
    {item: 'New York', isNumeric: false},
  ];

  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<TableRow cellData={mockCellData} isActive={true} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when isActive is true', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render when isActive is false', () => {
    wrapper.setProps({isActive: false});
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it('renders DataTable.Row component', () => {
    const dataTableRow = wrapper.find(DataTable.Row);
    expect(dataTableRow).toHaveLength(1);
    expect(dataTableRow.prop('style')).toEqual(
      expect.objectContaining(styles.tableRow),
    );
  });

  it('renders DataTable.Cell components with correct props', () => {
    const dataTableCells = wrapper.find(DataTable.Cell);
    expect(dataTableCells).toHaveLength(mockCellData.length);

    mockCellData.forEach((cell, index) => {
      const dataTableCell = dataTableCells.at(index);
      expect(dataTableCell.prop('numeric')).toBe(cell.isNumeric);
      expect(dataTableCell.children().text()).toBe(cell.item);
    });
  });
});
