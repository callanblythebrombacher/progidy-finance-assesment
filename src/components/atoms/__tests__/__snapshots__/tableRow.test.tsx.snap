// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow Atom renders correctly when isActive is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "borderBottomWidth": 0.5,
          "borderStyle": "solid",
          "minHeight": 48,
          "paddingHorizontal": 16,
        },
        {
          "borderBottomColor": "rgba(231, 224, 236, 1)",
        },
        {
          "backgroundColor": "white",
          "height": 70,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          false,
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
        testID="undefined-text-container"
      >
        John Doe
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          false,
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            },
            {
              "justifyContent": "flex-end",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
        testID="undefined-text-container"
      >
        25
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          false,
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
        testID="undefined-text-container"
      >
        New York
      </Text>
    </View>
  </View>
</View>
`;
